images/image-alt-text-context-form.html

```diff
@@ -0,0 +1,164 @@
+                        const altText = tinymce.DOM.getAttrib(node, 'alt');
+
+                        // Return the the alt text to be displayed as initial value
+                        return altText;
+                    },
+                    // Define the buttons associated with the context form, in this case
+                    // a save and cancel button
+                    commands: [
+                        {
+                            type: 'contextformbutton',
+                            primary: true,
+                            tooltip: 'Save',
+                            icon: 'checkmark',
+                            onAction: (contextFormApi) => {
+                                // Working directly with the DOM requires manually adding
+                                // the actions to the undo stack.
+                                // https://www.tiny.cloud/docs/api/tinymce/tinymce.undomanager/
+                                editor.undoManager.transact(() => {
+                                    // Get the selected element (in this case our image)
+                                    // https://www.tiny.cloud/docs/api/tinymce.dom/tinymce.dom.selection/#getnode
+                                    const node = editor.selection.getNode();
+
+                                    // Set the alt attribute on the selected image
+                                    // https://www.tiny.cloud/docs/api/tinymce.dom/tinymce.dom.domutils/#setattrib
+                                    const alt = tinymce.DOM.setAttrib(node, 'alt', contextFormApi.getValue());
+
+                                    // Close the context form
+                                    // https://www.tiny.cloud/docs/ui-components/contextform/#formapi
+                                    contextform.hide();
+                                });
+                            }
+                        },
+                        {
+                            type: 'contextformbutton',
+                            tooltip: 'Cancel',
+                            icon: 'close',
+                            onAction: (contextform) => {
+                                // The cancel button simply closes the context form
+                                // https://www.tiny.cloud/docs/ui-components/contextform/#formapi
+                                contextform.hide();
+                            }
+                        },
+                    ]
+                });
+            }
+        });
+    </script>
+    <style>
+        body {
+            margin: 2rem .5rem;
+            font-family: -apple-system, BlinkMacSystemFont, 'Segoe UI', Roboto, Oxygen-Sans, Ubuntu, Cantarell, 'Helvetica Neue', Helvetica, Arial, sans-serif;
+        }
+
+        main {
+            margin: 2rem auto;
+            max-width: 780px;
+        }
+    </style>
+</head>
+<body>
+<main>
+    <textarea id="editor">
+        <h1>Yosemite</h1>
```

Comment:
Is it valid HTML to put these elements inside a textarea? Would we be better off using a div?

Reply:
It's technically plain text, however, it really should technically be encoded correctly as the following, but the browser will handle that normally. We do see issues from people not encoding this correctly from time to time and it's normally in code samples.
```
<textarea id="editor">
  &lt;h1&gt;Yosemite&lt;/h1&gt;
  &lt;p&gt;Lorem ipsum dolor sit amet, consectetur adipiscing elit. Vestibulum a molestie nisl, at sodales diam. Vivamus ullamcorper convallis odio, ut lacinia sem.&lt;/p&gt;
  &lt;img src=&quot;images/yosemite.jpeg&quot; alt=&quot;A winding road with Yosemite in focus&quot;/&gt;
</textarea>
```

Reply:
We do this in all our demos in the docs so we'll continue doing this here too until there is a joint effort adressing this everywhere.